packages/cli-kit/src/public/node/hooks/prerun.ts

```diff
@@ -81,3 +86,18 @@ function findAlias(aliases: string[]) {
     return existingAlias.replace(/:/g, ' ')
   }
 }
+
+function warnOnAvailableUpgrade() {
+  const cliDependency = '@shopify/cli'
+  const currentVersion = CLI_KIT_VERSION
+
+  // Check in the background, once daily
+  void checkForNewVersion(cliDependency, currentVersion, {cacheExpiryInHours: 24})
+
+  // Warn if we previously found a new version
+  const lastVersion = checkForCachedNewVersion(cliDependency, currentVersion)
+  if (lastVersion) {
+    const packageManager = packageManagerFromUserAgent()
+    outputInfo(getOutputUpdateCLIReminder(packageManager, lastVersion))
```

Comment:
I think `newVersion` makes more sense here:

```suggestion
  const newVersion = checkForCachedNewVersion(cliDependency, currentVersion)
  if (newVersion) {
    const packageManager = packageManagerFromUserAgent()
    outputInfo(getOutputUpdateCLIReminder(packageManager, newVersion))
```